### PR TITLE
TFAC-177/178/219/226 Create the UserImpactTable and mutations to modify the fields

### PR DESCRIPTION
**Pull request by amaliwanag2**

### CODE REVIEW

File: dynamodb/tables.json
Comment (kmjennison) [inline dynamodb/tables.json:44]: @amaliwanag2 This attribute should be `userId`.

File: graphql/database/userImpact/UserImpactModel.js
Comment (kmjennison) [inline graphql/database/userImpact/UserImpactModel.js:47]: Do we also need `pendingUserReferrals` to know the number of referrals to display?

File: graphql/database/userImpact/updateImpact.js
Comment (kmjennison) [inline graphql/database/userImpact/updateImpact.js:7]: Fix "tabId" here

File: graphql/database/userImpact/UserImpactModel.js
Comment (kmjennison) [inline graphql/database/userImpact/UserImpactModel.js:61]: And do we need a flag to know if the frontend should show a "congrats, you reached X impact"?

File: graphql/database/userImpact/updateImpact.js
Comment (kmjennison) [inline graphql/database/userImpact/updateImpact.js:12]: I see what you're saying about this one mutation doing many things :) It's fine to leave for now, though I agree it's easier to reason about mutations and schema when a mutation is responsible for only one action.

File: graphql/database/userImpact/__tests__/updateImpact.test.js
Comment (kmjennison) [inline graphql/database/userImpact/__tests__/updateImpact.test.js:13]: Unneeded eslint and comment?

File: graphql/database/userImpact/UserImpactModel.js
Comment (kmjennison) [inline graphql/database/userImpact/UserImpactModel.js:80]: Remove unrelated comment

**Review by kmjennison — CHANGES_REQUESTED** [review]
Changes are only for a few stray comments/docs. Left a couple of questions just to raise whether there's additional fields we need, but I'll defer to your planning on this, and you can merge this before adding those (if needed).

File: graphql/database/userImpact/UserImpactModel.js
Comment (amaliwanag2) [reply graphql/database/userImpact/UserImpactModel.js:61]: yup!

File: graphql/database/userImpact/UserImpactModel.js
Comment (amaliwanag2) [reply graphql/database/userImpact/UserImpactModel.js:47]: yeah I was assuming we'd deduce that based on impact but it makes sense to store both

File: dynamodb/tables.json
Comment (kmjennison) [inline dynamodb/tables.json:42]: Need this same info in serverless.yml.

**Review by kmjennison — CHANGES_REQUESTED** [review]
Excellent work. Only change: please add the new table to serverless.yml as well so we can get it live in the cloud. You can set a very low minimum read/write capacity (probably just 1 for each), given the scaling will be slow.

File: graphql/serverless.yml
Comment (amaliwanag2) [inline graphql/serverless.yml:156]: @kmjennison ah, thanks for catching

**Review by kmjennison — APPROVED** [review]

**Pull request merged**
